### DISCUSSION TITLE
guys i just fixed it. MTU wasn't my issue(VPN etc. all share the same value - 1500).

### DISCUSSION THREAD

**Discussion by DineshRavindraRaju** [microsoft/WSL#8108] [General]
guys i just fixed it. MTU wasn't my issue(VPN etc. all share the same value - 1500). 

Here is what i did:
1. Create a file: `sudo nano /etc/wsl.conf`
2. Put the following lines in the file in order to ensure the your DNS changes do not get blown away

```bash
[network]
generateResolvConf = false
```

3. In a cmd window, run `wsl --shutdown`
4. Restart WSL2
5. Create a file: `sudo nano /etc/resolv.conf`. If it exists, replace existing one with this new file.
6. Put the following line in the file

```bash
nameserver 1.1.1.1 # Or use your DNS server by running ipconfig /all
```

7. `sudo chattr -f +i /etc/resolv.conf` so that file don't get blown away too.
8. Repeat step 3 and 4. You will see git working fine now.
9. Now `ping www.google.com` should have some response.

[source](https://gist.github.com/coltenkrauter/608cfe02319ce60facd76373249b8ca6)

_Originally posted by @kilgarenone in https://github.com/microsoft/WSL/issues/5068#issuecomment-970193138_

**Comment by r1g023** [comment]
This worked, thank you! VPN was on and my MTU had a different number, following your instructions I'm 'online' through my VPN on WSL2, thank you! @DineshRavindraRaju
